### ISSUE TITLE
Get warning on wsl2 ubuntu18

### ISSUE THREAD

**Issue by kuonumber** [rapidsai/dask-cuda#816]
I think I successfully initiate the gpu cluster..
![image](https://user-images.githubusercontent.com/16162246/144155618-1e830471-9920-4602-80cf-0f92174f2172.png)

But I repeatedly get the warning showed below..

File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/pynvml/nvml.py", line 743, in _nvmlCheckReturn
    raise NVMLError(ret)
pynvml.nvml.NVMLError_NotSupported: Not Supported

I have tried the method mentioned #761 (export DASK_DISTRIBUTED__DIAGNOSTICS__NVML=False)
But the result still does same.

Could you help me to resolve this situation? THX

**Comment by pentschev** [comment]
Could you post the output of `conda list`, `nvidia-smi` and the complete traceback of the error?

**Comment by kuonumber** [comment]
Sorry for my late response..
And  I have no idea to post them neatly,  I try my best..

###  nvidia-smi
![image](https://user-images.githubusercontent.com/16162246/144772370-a134fed1-972b-4457-a9d1-782a573f9919.png)

###  conda env
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
abseil-cpp                20210324.2           h9c3ff4c_0    conda-forge
argcomplete               1.12.3             pyhd3eb1b0_0
argon2-cffi               20.1.0           py37h27cfd23_1
arrow-cpp                 1.0.1           py37h3116320_57_cuda    conda-forge
arrow-cpp-proc            3.0.0                      cuda    conda-forge
async_generator           1.10             py37h28b3542_0
attrs                     21.2.0             pyhd3eb1b0_0
aws-c-auth                0.6.5                h00e28c2_5    conda-forge
aws-c-cal                 0.5.12               h8c99da7_6    conda-forge
aws-c-common              0.6.16               h7f98852_0    conda-forge
aws-c-compression         0.2.14               h4f5cfdb_6    conda-forge
aws-c-event-stream        0.2.7               hedd7935_30    conda-forge
aws-c-http                0.6.10               he2927c9_1    conda-forge
aws-c-io                  0.10.13              h4e1a65d_2    conda-forge
aws-c-mqtt                0.7.8                h1e34fcd_9    conda-forge
aws-c-s3                  0.1.27               hbe77d09_8    conda-forge
aws-c-sdkutils            0.1.1                h4f5cfdb_3    conda-forge
aws-checksums             0.1.12               h4f5cfdb_5    conda-forge
aws-crt-cpp               0.17.4               ha10c5ac_2    conda-forge
aws-sdk-cpp               1.9.140              h94f6c54_1    conda-forge
backcall                  0.2.0              pyhd3eb1b0_0
bleach                    4.0.0              pyhd3eb1b0_0
bokeh                     2.1.1                    py37_0
brotli                    1.0.9                h7f98852_6    conda-forge
brotli-bin                1.0.9                h7f98852_6    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2021.10.8            ha878542_0    conda-forge
cachetools                4.2.4              pyhd8ed1ab_0    conda-forge
cffi                      1.14.5           py37h261ae71_0
click                     8.0.3              pyhd3eb1b0_0
cloudpickle               2.0.0              pyhd3eb1b0_0
cudatoolkit               11.0.221             h6bb024c_0    nvidia
cudf                      21.06.01        cuda_11.0_py37_g101fc0fda4_2    rapidsai
cudnn                     8.0.0                cuda11.0_0    nvidia
cupy                      8.0.0            py37h0ce7dbb_0    rapidsai
cycler                    0.11.0                   pypi_0    pypi
cytoolz                   0.11.0           py37h7b6447c_0
dask                      2021.5.1           pyhd3eb1b0_0
dask-core                 2021.5.1           pyhd3eb1b0_0
dask-cuda                 21.08.00                 py37_0    rapidsai
dask-cudf                 21.06.01        py37_g101fc0fda4_2    rapidsai
dbus                      1.13.18              hb2f20db_0
debugpy                   1.4.1            py37h295c915_0
decorator                 5.1.0              pyhd3eb1b0_0
defusedxml                0.7.1              pyhd3eb1b0_0
distributed               2021.5.1         py37h06a4308_0
dlpack                    0.5                  h9c3ff4c_0    conda-forge
entrypoints               0.3                      py37_0
expat                     2.4.1                h2531618_2
fastavro                  1.4.7            py37h5e8e339_1    conda-forge
fastrlock                 0.8              py37hcd2ae1e_1    conda-forge
fontconfig                2.13.1               h6c09931_0
fonttools                 4.28.2                   pypi_0    pypi
freetype                  2.11.0               h70c0345_0
fsspec                    2021.11.0          pyhd8ed1ab_0    conda-forge
gflags                    2.2.2             he1b5a44_1004    conda-forge
glib                      2.69.1               h5202010_0
glog                      0.5.0                h48cff8f_0    conda-forge
grpc-cpp                  1.41.1               h75e9d12_2    conda-forge
gst-plugins-base          1.14.0               h8213a91_2
gstreamer                 1.14.0               h28cd5cc_2
heapdict                  1.0.1              pyhd3eb1b0_0
icu                       58.2                 he6710b0_3
importlib-metadata        4.8.1            py37h06a4308_0
importlib_metadata        4.8.1                hd3eb1b0_0
install                   1.3.4                    pypi_0    pypi
ipykernel                 6.4.1            py37h06a4308_1
ipython                   7.29.0           py37hb070fc8_0
ipython_genutils          0.2.0              pyhd3eb1b0_1
ipywidgets                7.6.5              pyhd3eb1b0_1
jedi                      0.18.0           py37h06a4308_1
jinja2                    3.0.2              pyhd3eb1b0_0
jpeg                      9d                   h7f8727e_0
jsonschema                3.2.0              pyhd3eb1b0_2
jupyter                   1.0.0                    py37_7
jupyter-contrib-core      0.3.3                    pypi_0    pypi
jupyter-contrib-nbextensions 0.5.1                    pypi_0    pypi
jupyter-highlight-selected-word 0.2.0                    pypi_0    pypi
jupyter-latex-envs        1.4.6                    pypi_0    pypi
jupyter-nbextensions-configurator 0.4.1                    pypi_0    pypi
jupyter_client            7.0.1              pyhd3eb1b0_0
jupyter_console           6.4.0              pyhd3eb1b0_0
jupyter_core              4.9.1            py37h06a4308_0
jupyterlab_pygments       0.1.2                      py_0
jupyterlab_widgets        1.0.0              pyhd3eb1b0_1
kiwisolver                1.3.2                    pypi_0    pypi
krb5                      1.19.2               hcc1bbae_3    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libblas                   3.9.0           12_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h7f98852_6    conda-forge
libbrotlidec              1.0.9                h7f98852_6    conda-forge
libbrotlienc              1.0.9                h7f98852_6    conda-forge
libcblas                  3.9.0           12_linux64_openblas    conda-forge
libcudf                   21.06.01        cuda11.0_g101fc0fda4_2    rapidsai
libcurl                   7.80.0               h2574ce0_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               h9b69904_4    conda-forge
libffi                    3.3                  he6710b0_2
libgcc-ng                 11.2.0              h1d223b6_11    conda-forge
libgfortran-ng            11.2.0              h69a702a_11    conda-forge
libgfortran5              11.2.0              h5c6108e_11    conda-forge
libgomp                   11.2.0              h1d223b6_11    conda-forge
liblapack                 3.9.0           12_linux64_openblas    conda-forge
libnghttp2                1.43.0               h812cca2_1    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.18          pthreads_h8fe5266_0    conda-forge
libpng                    1.6.37               hbc83047_0
libprotobuf               3.18.1               h780b84a_0    conda-forge
librmm                    21.06.00        cuda11.0_gee432a0_0    rapidsai
libsodium                 1.0.18               h7b6447c_0
libssh2                   1.10.0               ha56f1ee_2    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_11    conda-forge
libthrift                 0.15.0               he6d91bd_1    conda-forge
libtiff                   4.0.9                he85c1e1_2
libutf8proc               2.6.1                h7f98852_0    conda-forge
libuuid                   1.0.3                h7f8727e_2
libxcb                    1.14                 h7b6447c_0
libxml2                   2.9.12               h03d6c58_0
libzlib                   1.2.11            h36c2ea0_1013    conda-forge
llvmlite                  0.37.0           py37he1b5a44_0    numba
locket                    0.2.1            py37h06a4308_1
lxml                      4.6.4                    pypi_0    pypi
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
markupsafe                2.0.1            py37h27cfd23_0
matplotlib                3.5.0                    pypi_0    pypi
matplotlib-inline         0.1.2              pyhd3eb1b0_2
mistune                   0.8.4           py37h14c3975_1001
msgpack-python            1.0.2            py37hff7bd54_1
nbclient                  0.5.3              pyhd3eb1b0_0
nbconvert                 5.6.1                    py37_1
nbformat                  5.1.3              pyhd3eb1b0_0
nccl                      2.7.8.1            h4962215_100    nvidia
ncurses                   6.2                  h58526e2_4    conda-forge
nest-asyncio              1.5.1              pyhd3eb1b0_0
notebook                  6.4.5            py37h06a4308_0
numba                     0.54.0rc1       np1.11py3.7h04863e7_g9bed2ebb2_0    numba
numpy                     1.21.4           py37h31617e3_0    conda-forge
nvtx                      0.2.3            py37h5e8e339_0    conda-forge
olefile                   0.46                     py37_0
openssl                   1.1.1l               h7f98852_0    conda-forge
orc                       1.7.1                h68e2c4e_0    conda-forge
packaging                 21.0               pyhd8ed1ab_0    conda-forge
pandas                    1.2.5            py37h219a48f_0    conda-forge
pandoc                    2.12                 h06a4308_0
pandocfilters             1.4.3            py37h06a4308_1
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.8.2              pyhd3eb1b0_0
partd                     1.2.0              pyhd3eb1b0_0
pcre                      8.45                 h295c915_0
pexpect                   4.8.0              pyhd3eb1b0_3
pickleshare               0.7.5           pyhd3eb1b0_1003
pillow                    8.4.0                    pypi_0    pypi
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
prometheus_client         0.11.0             pyhd3eb1b0_0
prompt-toolkit            3.0.20             pyhd3eb1b0_0
prompt_toolkit            3.0.20               hd3eb1b0_0
protobuf                  3.18.1           py37hcd2ae1e_0    conda-forge
psutil                    5.8.0            py37h27cfd23_1
ptyprocess                0.7.0              pyhd3eb1b0_2
pyarrow                   1.0.1           py37h08c6592_57_cuda    conda-forge
pycparser                 2.21               pyhd3eb1b0_0
pygments                  2.10.0             pyhd3eb1b0_0
pynvml                    11.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.6              pyhd8ed1ab_0    conda-forge
pyqt                      5.9.2            py37h05f1152_2
pyrsistent                0.18.0           py37heee7806_0
python                    3.7.11               h12debd9_0
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.7                     2_cp37m    conda-forge
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.2.1           py37h295c915_1
qt                        5.9.7                h5867ecd_1
qtconsole                 5.1.1              pyhd3eb1b0_0
qtpy                      1.10.0             pyhd3eb1b0_0
re2                       2021.11.01           h9c3ff4c_0    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
rmm                       21.06.00        cuda_11.0_py37_gee432a0_0    rapidsai
s2n                       1.2.0                h9b69904_0    conda-forge
send2trash                1.8.0              pyhd3eb1b0_1
setuptools                59.1.0           py37h89c1867_0    conda-forge
setuptools-scm            6.3.2                    pypi_0    pypi
sip                       4.19.8           py37hf484d3e_0
six                       1.16.0             pyh6c4a22f_0    conda-forge

### trackback
tornado.application - ERROR - Exception in callback <bound method SystemMonitor.update of <SystemMonitor: cpu: 22 memory: 261 MB fds: 28>>
Traceback (most recent call last):
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/tornado/ioloop.py", line 905, in _run
    return self.callback()
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/distributed/system_monitor.py", line 99, in update
    gpu_metrics = nvml.real_time()
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/distributed/diagnostics/nvml.py", line 38, in real_time
    "utilization": pynvml.nvmlDeviceGetUtilizationRates(h).gpu,
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/pynvml/nvml.py", line 2061, in nvmlDeviceGetUtilizationRates
    _nvmlCheckReturn(ret)
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/pynvml/nvml.py", line 743, in _nvmlCheckReturn
    raise NVMLError(ret)
pynvml.nvml.NVMLError_NotSupported: Not Supported
tornado.application - ERROR - Exception in callback <bound method SystemMonitor.update of <SystemMonitor: cpu: 27 memory: 271 MB fds: 28>>
Traceback (most recent call last):
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/tornado/ioloop.py", line 905, in _run
    return self.callback()
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/distributed/system_monitor.py", line 99, in update
    gpu_metrics = nvml.real_time()
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/distributed/diagnostics/nvml.py", line 38, in real_time
    "utilization": pynvml.nvmlDeviceGetUtilizationRates(h).gpu,
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/pynvml/nvml.py", line 2061, in nvmlDeviceGetUtilizationRates
    _nvmlCheckReturn(ret)
  File "/home/kuo/miniconda3/envs/cudf-env/lib/python3.7/site-packages/pynvml/nvml.py", line 743, in _nvmlCheckReturn
    raise NVMLError(ret)
pynvml.nvml.NVMLError_NotSupported: Not Supported

**Comment by pentschev** [comment]
> And I have no idea to post them neatly, I try my best..

That's ok, we can still understand it! 🙂 

For future reference, if you want to format code nicely, feel free to refer to [GitHub's quoting code instructions](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code).

Thanks for the output, I see you're using RAPIDS 21.06, that's too old and from a time when `DASK_DISTRIBUTED__DIAGNOSTICS__NVML` didn't exist yet. I recommend trying the same again with RAPIDS 21.10, you can find instructions to install a more recent version in the [RAPIDS Release Selector](https://rapids.ai/start.html#get-rapids). Could you try upgrading and checking if that works?

**Comment by kuonumber** [comment]
Ha, thank you for github tips, I refined the  previous post.
I created a new env and installed dependencies (using command from RAPIDS Release Selector).
But I got a different error.  =\
```
---------------------------------------------------------------------------
CUDARuntimeError                          Traceback (most recent call last)
/tmp/ipykernel_277/1669699230.py in <module>
----> 1 cupy.cuda.Device(1).mem_info

cupy/cuda/device.pyx in cupy.cuda.device.Device.mem_info.__get__()

cupy/cuda/device.pyx in cupy.cuda.device.Device.mem_info.__get__()

cupy_backends/cuda/api/runtime.pyx in cupy_backends.cuda.api.runtime.memGetInfo()

cupy_backends/cuda/api/runtime.pyx in cupy_backends.cuda.api.runtime.check_status()

CUDARuntimeError: cudaErrorNotSupported: operation not supported
```
### nvidia-smi
```
Tue Dec  7 16:14:37 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.00       Driver Version: 510.06       CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  On   | 00000000:01:00.0 Off |                  N/A |
| N/A   36C    P8    N/A /  N/A |    146MiB /  4096MiB |     N/A      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce ...  On   | 00000000:09:00.0 Off |                  N/A |
|  0%   33C    P8    10W / 250W |    300MiB / 11264MiB |     N/A      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```
### conda list
```
# packages in environment at /home/kuo/miniconda3/envs/rapids-21.10:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
abseil-cpp                20210324.2           h9c3ff4c_0    conda-forge
aiohttp                   3.8.1            py37h5e8e339_0    conda-forge
aiosignal                 1.2.0              pyhd8ed1ab_0    conda-forge
anyio                     3.4.0            py37h89c1867_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
argcomplete               1.12.3             pyhd8ed1ab_2    conda-forge
argon2-cffi               21.1.0           py37h5e8e339_2    conda-forge
arrow-cpp                 5.0.0           py37h5a62f8a_4_cuda    conda-forge
arrow-cpp-proc            3.0.0                      cuda    conda-forge
async-timeout             4.0.1              pyhd8ed1ab_0    conda-forge
async_generator           1.10                       py_0    conda-forge
asynctest                 0.13.0                     py_0    conda-forge
attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
aws-c-cal                 0.5.11               h95a6274_0    conda-forge
aws-c-common              0.6.2                h7f98852_0    conda-forge
aws-c-event-stream        0.2.7               h3541f99_13    conda-forge
aws-c-io                  0.10.5               hfb6a706_0    conda-forge
aws-checksums             0.1.11               ha31a3da_7    conda-forge
aws-sdk-cpp               1.8.186              hb4091e7_3    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
blazingsql                21.10.0                  pypi_0    pypi
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
blosc                     1.21.0               h9c3ff4c_0    conda-forge
bokeh                     2.4.0            py37h89c1867_0    conda-forge
boost                     1.72.0           py37h48f8a5e_1    conda-forge
boost-cpp                 1.72.0               h312852a_5    conda-forge
brotli                    1.0.9                h7f98852_6    conda-forge
brotli-bin                1.0.9                h7f98852_6    conda-forge
brotlipy                  0.7.0           py37h5e8e339_1003    conda-forge
brunsli                   0.1                  h9c3ff4c_0    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
c-blosc2                  2.0.4                h5f21a17_1    conda-forge
ca-certificates           2021.10.8            ha878542_0    conda-forge
cachetools                4.2.4              pyhd8ed1ab_0    conda-forge
cairo                     1.16.0            h6cf1ce9_1008    conda-forge
certifi                   2021.10.8        py37h89c1867_1    conda-forge
cffi                      1.15.0           py37h036bc23_0    conda-forge
cfitsio                   3.470                hb418390_7    conda-forge
charls                    2.2.0                h9c3ff4c_0    conda-forge
charset-normalizer        2.0.9              pyhd8ed1ab_0    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               2.0.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
colorcet                  3.0.0              pyhd8ed1ab_0    conda-forge
cryptography              36.0.0           py37hf1a17b8_0    conda-forge
cucim                     21.10.00        cuda_11.0_py37_gd7ac21f_0    rapidsai
cudatoolkit               11.0.221             h6bb024c_0    nvidia
cudf                      21.10.01        cuda_11.0_py37_ga1d2d13a14_0    rapidsai
cudf_kafka                21.10.01        py37_ga1d2d13a14_0    rapidsai
cugraph                   21.10.00        cuda11.0_py37_g84617024_0    rapidsai
cuml                      21.10.02        cuda11.0_py37_gcd9251271_0    rapidsai
cupy                      9.6.0            py37h580ddce_0    conda-forge
curl                      7.80.0               h2574ce0_0    conda-forge
cusignal                  21.10.00        py37_gff14a10_0    rapidsai
cuspatial                 21.10.00        py37_gba20298_0    rapidsai
custreamz                 21.10.01        py37_ga1d2d13a14_0    rapidsai
cuxfilter                 21.10.00        py37_g003d3d6_0    rapidsai
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cyrus-sasl                2.1.27               h230043b_5    conda-forge
cytoolz                   0.11.2           py37h5e8e339_1    conda-forge
dask                      2021.9.1           pyhd8ed1ab_0    conda-forge
dask-core                 2021.9.1           pyhd8ed1ab_0    conda-forge
dask-cuda                 21.10.00                 py37_0    rapidsai
dask-cudf                 21.10.01        py37_ga1d2d13a14_0    rapidsai
datashader                0.11.1             pyh9f0ad1d_0    conda-forge
datashape                 0.5.4                      py_1    conda-forge
debugpy                   1.5.1            py37hcd2ae1e_0    conda-forge
decorator                 5.1.0              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
distributed               2021.9.1         py37h89c1867_0    conda-forge
dlpack                    0.5                  h9c3ff4c_0    conda-forge
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
expat                     2.4.1                h9c3ff4c_0    conda-forge
faiss-proc                1.0.0                      cuda    rapidsai
fastavro                  1.4.7            py37h5e8e339_1    conda-forge
fastrlock                 0.8              py37hcd2ae1e_1    conda-forge
fiona                     1.8.20           py37hb7e2723_1    conda-forge
fontconfig                2.13.1            hba837de_1005    conda-forge
fonttools                 4.28.3           py37h5e8e339_0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
freexl                    1.0.6                h7f98852_0    conda-forge
frozenlist                1.2.0            py37h5e8e339_1    conda-forge
fsspec                    2021.11.1          pyhd8ed1ab_0    conda-forge
future                    0.18.2           py37h89c1867_4    conda-forge
gdal                      3.3.1            py37hd5a0ba4_3    conda-forge
geopandas                 0.9.0              pyhd8ed1ab_1    conda-forge
geopandas-base            0.9.0              pyhd8ed1ab_1    conda-forge
geos                      3.9.1                h9c3ff4c_2    conda-forge
geotiff                   1.6.0                h4f31c25_6    conda-forge
gettext                   0.19.8.1          h73d1719_1008    conda-forge
gflags                    2.2.2             he1b5a44_1004    conda-forge
giflib                    5.2.1                h36c2ea0_2    conda-forge
glog                      0.5.0                h48cff8f_0    conda-forge
google-cloud-cpp          1.29.0               hb967e95_1    conda-forge
greenlet                  1.1.2            py37hcd2ae1e_1    conda-forge
grpc-cpp                  1.39.1               h850795e_1    conda-forge
hdf4                      4.2.15               h10796ff_3    conda-forge
hdf5                      1.12.1          nompi_h2750804_103    conda-forge
heapdict                  1.0.1                      py_0    conda-forge
icu                       68.2                 h9c3ff4c_0    conda-forge
idna                      3.1                pyhd3deb0d_0    conda-forge
imagecodecs               2021.8.26        py37hfe5a812_1    conda-forge
imageio                   2.13.1             pyhd8ed1ab_1    conda-forge
importlib-metadata        4.8.2            py37h89c1867_0    conda-forge
importlib_metadata        4.8.2                hd8ed1ab_0    conda-forge
importlib_resources       5.4.0              pyhd8ed1ab_0    conda-forge
ipykernel                 6.6.0            py37h6531663_0    conda-forge
ipython                   7.30.1           py37h89c1867_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.5              pyhd8ed1ab_0    conda-forge
jbig                      2.1               h7f98852_2003    conda-forge
jedi                      0.18.1           py37h89c1867_0    conda-forge
jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
joblib                    1.1.0              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   h36c2ea0_0    conda-forge
jpype1                    1.3.0            py37h2527ec5_2    conda-forge
json-c                    0.15                 h98cffda_0    conda-forge
jsonschema                4.2.1              pyhd8ed1ab_0    conda-forge
jupyter-server-proxy      3.2.0              pyhd8ed1ab_0    conda-forge
jupyter_client            7.1.0              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.1            py37h89c1867_1    conda-forge
jupyter_server            1.12.1             pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_widgets        1.0.2              pyhd8ed1ab_0    conda-forge
jxrlib                    1.1                  h7f98852_2    conda-forge
kealib                    1.4.14               h87e4c3c_3    conda-forge
kiwisolver                1.3.2            py37h2527ec5_1    conda-forge
krb5                      1.19.2               hcc1bbae_3    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
lerc                      3.0                  h9c3ff4c_0    conda-forge
libaec                    1.0.6                h9c3ff4c_0    conda-forge
libblas                   3.9.0           12_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h7f98852_6    conda-forge
libbrotlidec              1.0.9                h7f98852_6    conda-forge
libbrotlienc              1.0.9                h7f98852_6    conda-forge
libcblas                  3.9.0           12_linux64_openblas    conda-forge
libcrc32c                 1.1.2                h9c3ff4c_0    conda-forge
libcucim                  21.10.00        cuda11.0_gd7ac21f_0    rapidsai
libcudf                   21.10.01        cuda11.0_ga1d2d13a14_0    rapidsai
libcudf_kafka             21.10.01          ga1d2d13a14_0    rapidsai
libcugraph                21.10.00        cuda11.0_g84617024_0    rapidsai
libcuml                   21.10.02        cuda11.0_gcd9251271_0    rapidsai
libcumlprims              21.10.00        cuda11.0_g167dc59_0    nvidia
libcurl                   7.80.0               h2574ce0_0    conda-forge
libcuspatial              21.10.00        cuda11.0_gba20298_0    rapidsai
libdap4                   3.20.6               hd7c4107_2    conda-forge
libdeflate                1.8                  h7f98852_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               h9b69904_4    conda-forge
libfaiss                  1.7.0           cuda110h8045045_8_cuda    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0              h1d223b6_11    conda-forge
libgcrypt                 1.9.4                h7f98852_0    conda-forge
libgdal                   3.3.1                h6214c1d_3    conda-forge
libgfortran-ng            11.2.0              h69a702a_11    conda-forge
libgfortran5              11.2.0              h5c6108e_11    conda-forge
libglib                   2.70.1               h174f98d_0    conda-forge
libgomp                   11.2.0              h1d223b6_11    conda-forge
libgpg-error              1.42                 h9c3ff4c_0    conda-forge
libgsasl                  1.10.0               h5b4c23d_0    conda-forge
libhwloc                  2.3.0                h5e5b7d1_1    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
libkml                    1.3.0             hd79254b_1012    conda-forge
liblapack                 3.9.0           12_linux64_openblas    conda-forge
libllvm10                 10.0.1               he513fc3_3    conda-forge
libnetcdf                 4.8.1           nompi_hb3fd0d9_101    conda-forge
libnghttp2                1.43.0               h812cca2_1    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libntlm                   1.4               h7f98852_1002    conda-forge
libopenblas               0.3.18          pthreads_h8fe5266_0    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libpq                     13.5                 hd57d9b9_0    conda-forge
libprotobuf               3.16.0               h780b84a_0    conda-forge
librdkafka                1.6.1                hc49e61c_1    conda-forge
librmm                    21.10.01        cuda11.0_gc54767f_0    rapidsai
librttopo                 1.1.0                h1185371_6    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libspatialindex           1.9.3                h9c3ff4c_4    conda-forge
libspatialite             5.0.1                h8694cbe_6    conda-forge
libssh2                   1.10.0               ha56f1ee_2    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_11    conda-forge
libthrift                 0.14.2               he6d91bd_1    conda-forge
libtiff                   4.3.0                h6f004c6_2    conda-forge
libutf8proc               2.6.1                h7f98852_0    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libuv                     1.42.0               h7f98852_0    conda-forge
libwebp                   1.2.1                h3452ae3_0    conda-forge
libwebp-base              1.2.1                h7f98852_0    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxgboost                1.4.2dev.rapidsai21.10      cuda11.0_0    rapidsai
libxml2                   2.9.12               h72842e0_0    conda-forge
libzip                    1.8.0                h4de3113_1    conda-forge
libzlib                   1.2.11            h36c2ea0_1013    conda-forge
libzopfli                 1.0.3                h9c3ff4c_0    conda-forge
llvmlite                  0.36.0           py37h9d7f4d0_0    conda-forge
locket                    0.2.0                      py_2    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
mapclassify               2.4.3              pyhd8ed1ab_0    conda-forge
markdown                  3.3.6              pyhd8ed1ab_0    conda-forge
markupsafe                2.0.1            py37h5e8e339_1    conda-forge
matplotlib-base           3.5.0            py37h1058ff1_0    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mistune                   0.8.4           py37h5e8e339_1005    conda-forge
msgpack-python            1.0.3            py37h2527ec5_0    conda-forge
multidict                 5.2.0            py37h5e8e339_1    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
nbclient                  0.5.9              pyhd8ed1ab_0    conda-forge
nbconvert                 6.3.0            py37h89c1867_1    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
nccl                      2.11.4.1             h96e36e3_0    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
netifaces                 0.10.9          py37h5e8e339_1004    conda-forge
networkx                  2.6.3              pyhd8ed1ab_1    conda-forge
nlohmann_json             3.9.1                h9c3ff4c_1    conda-forge
nodejs                    14.17.4              h92b4a50_0    conda-forge
notebook                  6.4.6              pyha770c72_0    conda-forge
numba                     0.53.1           py37hb11d6e1_1    conda-forge
numpy                     1.21.4           py37h31617e3_0    conda-forge
nvtx                      0.2.3            py37h5e8e339_1    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openjdk                   8.0.312              h7f98852_0    conda-forge
openjpeg                  2.4.0                hb52868f_1    conda-forge
openssl                   1.1.1l               h7f98852_0    conda-forge
orc                       1.6.10               h58a87f1_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.3.4            py37he8f5f7f_1    conda-forge
pandoc                    2.16.2               h7f98852_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
panel                     0.12.4             pyhd8ed1ab_0    conda-forge
param                     1.12.0             pyh6c4a22f_0    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.2.0              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 h9c3ff4c_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickle5                   0.0.12           py37h5e8e339_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.4.0            py37h0f21c89_0    conda-forge
pip                       21.3.1             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h36c2ea0_0    conda-forge
pooch                     1.5.2              pyhd8ed1ab_0    conda-forge
poppler                   21.03.0              h93df280_0    conda-forge
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
postgresql                13.5                 h2510834_0    conda-forge
proj                      8.0.1                h277dcde_0    conda-forge
prometheus_client         0.12.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.23             pyha770c72_0    conda-forge
protobuf                  3.16.0           py37hcd2ae1e_0    conda-forge
psutil                    5.8.0            py37h5e8e339_2    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure-sasl                 0.6.2              pyhd8ed1ab_0    conda-forge
py-xgboost                1.4.2dev.rapidsai21.10  cuda11.0py37_0    rapidsai
pyarrow                   5.0.0           py37hf0016df_4_cuda    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyct                      0.4.6                      py_0    conda-forge
pyct-core                 0.4.6                      py_0    conda-forge
pydeck                    0.5.0              pyh9f0ad1d_0    conda-forge
pyee                      8.1.0              pyh9f0ad1d_0    conda-forge
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pyhive                    0.6.4              pyhd8ed1ab_0    conda-forge
pynvml                    11.0.0             pyhd8ed1ab_0    conda-forge
pyopenssl                 21.0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.6              pyhd8ed1ab_0    conda-forge
pyppeteer                 0.2.6              pyhd8ed1ab_0    conda-forge
pyproj                    3.1.0            py37h1939ec9_4    conda-forge
pyrsistent                0.18.0           py37h5e8e339_0    conda-forge
pysocks                   1.7.1            py37h89c1867_4    conda-forge
python                    3.7.12          hb7a2778_100_cpython    conda-forge
python-confluent-kafka    1.6.0            py37h5e8e339_1    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.7                     2_cp37m    conda-forge
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
pyviz_comms               2.1.0              pyhd8ed1ab_0    conda-forge
pywavelets                1.2.0            py37hb1e94ed_1    conda-forge
pyyaml                    6.0              py37h5e8e339_3    conda-forge
pyzmq                     22.3.0           py37h336d617_1    conda-forge
rapids                    21.10.00        cuda11.0_py37_ge66f011_114    rapidsai
rapids-blazing            21.10.00        cuda11.0_py37_ge66f011_114    rapidsai
rapids-xgboost            21.10.00        cuda11.0_py37_ge66f011_114    rapidsai
re2                       2021.09.01           h9c3ff4c_0    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
requests                  2.26.0             pyhd8ed1ab_1    conda-forge
rmm                       21.10.01        cuda_11.0_py37_gc54767f_0    rapidsai
rtree                     0.9.7            py37h0b55af0_3    conda-forge
s2n                       1.0.10               h9b69904_0    conda-forge
scikit-image              0.18.1           py37hdc94413_0    conda-forge
scikit-learn              1.0.1            py37hf9e9bfc_2    conda-forge
scipy                     1.7.3            py37hf2a6cf1_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                49.6.0           py37h89c1867_3    conda-forge
shapely                   1.8.0            py37h48c49eb_0    conda-forge
simpervisor               0.4                pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.1.8                he1b5a44_3    conda-forge
sniffio                   1.2.0            py37h89c1867_2    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
spdlog                    1.8.5                h4bd325d_0    conda-forge
sqlalchemy                1.4.27           py37h5e8e339_0    conda-forge
sqlite                    3.37.0               h9cd32fc_0    conda-forge
streamz                   0.6.3              pyh6c4a22f_0    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
terminado                 0.12.1           py37h89c1867_1    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.0.0              pyh8a188c0_0    conda-forge
thrift                    0.15.0           py37hcd2ae1e_1    conda-forge
thrift_sasl               0.4.3              pyhd8ed1ab_1    conda-forge
tifffile                  2021.11.2          pyhd8ed1ab_0    conda-forge
tiledb                    2.3.4                he87e0bf_0    conda-forge
tk                        8.6.11               h27826a3_1    conda-forge
toolz                     0.11.2             pyhd8ed1ab_0    conda-forge
tornado                   6.1              py37h5e8e339_2    conda-forge
tqdm                      4.62.3             pyhd8ed1ab_0    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
treelite                  2.1.0            py37h4b3d254_0    conda-forge
treelite-runtime          2.1.0                    pypi_0    pypi
typing-extensions         4.0.1                hd8ed1ab_0    conda-forge
typing_extensions         4.0.1              pyha770c72_0    conda-forge
tzcode                    2021e                h7f98852_0    conda-forge
tzdata                    2021e                he74cb21_0    conda-forge
ucx                       1.11.1+gc58db6b      cuda11.0_0    rapidsai
ucx-proc                  1.0.0                       gpu    rapidsai
ucx-py                    0.22.0          py37_gc58db6b_0    rapidsai
unicodedata2              13.0.0.post2     py37h5e8e339_4    conda-forge
urllib3                   1.26.7             pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          1.2.1            py37h89c1867_0    conda-forge
websockets                9.1              py37h5e8e339_0    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
widgetsnbextension        3.5.2            py37h89c1867_1    conda-forge
xarray                    0.20.1             pyhd8ed1ab_0    conda-forge
xerces-c                  3.2.3                h9d8b166_3    conda-forge
xgboost                   1.4.2dev.rapidsai21.10  cuda11.0py37_0    rapidsai
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libice               1.0.10               h7f98852_0    conda-forge
xorg-libsm                1.2.3             hd9c2040_1000    conda-forge
xorg-libx11               1.7.2                h7f98852_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxrender           0.9.10            h7f98852_1003    conda-forge
xorg-renderproto          0.11.1            h7f98852_1002    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
yaml                      0.2.5                h516909a_0    conda-forge
yarl                      1.7.2            py37h5e8e339_1    conda-forge
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
zfp                       0.5.5                h9c3ff4c_8    conda-forge
zict                      2.0.0                      py_0    conda-forge
zipp                      3.6.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11            h36c2ea0_1013    conda-forge
zstd                      1.5.0                ha95c52a_0    conda-forge
```

**Comment by pentschev** [comment]
Ok, now the first error you're hitting is the issue that was fixed in https://github.com/dask/distributed/pull/5469 . You have two options:

1. Choose a newer CUDA toolkit (for example, 11.2);
2. Upgrade to RAPIDS 21.12 (even though it's still the development).

I believe either one of the options above should work around that first error.

The second error isn't clear to me if it's a side effect of the first or something different, so we'll have to see if the above resolves everything. However, I noticed now that you have two very different GPUs, a 4GB one and an 11GB one. I must warn you that GPUs with different total memory sizes isn't something we actively test and support, therefore we can't guarantee everything will work smoothly with that GPU configuration. It's also not clear which GPUs you have, that 4GB in particular may be something that isn't supported at all, could you post the output of `nvidia-smi -L` too? This will print the GPU type as well.

**Comment by kuonumber** [comment]
One is on my laptop, the other is on razer core x (egpu).

GPU 0: NVIDIA GeForce GTX 1050 Ti (UUID: GPU-94045f6c-1e8c-6875-e2ea-edcf51ed7683)
GPU 1: NVIDIA GeForce GTX 1080 Ti (UUID: GPU-4cdd4290-b80a-3033-296e-aa4e4320d3f0)

**Comment by pentschev** [comment]
I don't think we've ever tested eGPUs, so I can't guarantee this will work either but it will be interesting to know what happens there. Let us know whether any of the options I posted above will let you get past those errors.

**Comment by charlesbluca** [comment]
I can recreate this issue using the latest of dask-cuda, distributed, and cudatoolkit 11.5:

```python
from dask_cuda import LocalCUDACluster
cluster = LocalCUDACluster()
```
```
/home/charlesb/dev/rapids-dask/dask-cuda/dask_cuda/utils.py:217: UserWarning: Cannot get CPU affinity for device with index 0, setting default affinity
  warnings.warn(
/home/charlesb/dev/rapids-dask/dask-cuda/dask_cuda/utils.py:217: UserWarning: Cannot get CPU affinity for device with index 1, setting default affinity
  warnings.warn(
distributed.preloading - INFO - Import preload module: dask_cuda.initialize
Unable to start CUDA Context
Traceback (most recent call last):
  File "/home/charlesb/dev/rapids-dask/dask-cuda/dask_cuda/initialize.py", line 32, in _create_cuda_context
    distributed.comm.ucx.init_once()
  File "/home/charlesb/dev/rapids-dask/distributed/distributed/comm/ucx.py", line 86, in init_once
    pre_existing_cuda_context = has_cuda_context()
  File "/home/charlesb/dev/rapids-dask/distributed/distributed/diagnostics/nvml.py", line 78, in has_cuda_context
    running_processes = pynvml.nvmlDeviceGetComputeRunningProcesses_v2(handle)
  File "/home/charlesb/miniconda3/envs/rapids-dask/lib/python3.8/site-packages/pynvml/nvml.py", line 2141, in nvmlDeviceGetComputeRunningProcesses_v2
    raise NVMLError(ret)
pynvml.nvml.NVMLError_Unknown: Unknown Error
distributed.preloading - INFO - Import preload module: dask_cuda.initialize
Unable to start CUDA Context
Traceback (most recent call last):
  File "/home/charlesb/dev/rapids-dask/dask-cuda/dask_cuda/initialize.py", line 32, in _create_cuda_context
    distributed.comm.ucx.init_once()
  File "/home/charlesb/dev/rapids-dask/distributed/distributed/comm/ucx.py", line 86, in init_once
    pre_existing_cuda_context = has_cuda_context()
  File "/home/charlesb/dev/rapids-dask/distributed/distributed/diagnostics/nvml.py", line 78, in has_cuda_context
    running_processes = pynvml.nvmlDeviceGetComputeRunningProcesses_v2(handle)
  File "/home/charlesb/miniconda3/envs/rapids-dask/lib/python3.8/site-packages/pynvml/nvml.py", line 2141, in nvmlDeviceGetComputeRunningProcesses_v2
    raise NVMLError(ret)
pynvml.nvml.NVMLError_Unknown: Unknown Error
```

<details>
<summary>Environment</summary>

```
$ conda list "dask|distributed|cuda"
# packages in environment at /home/charlesb/miniconda3/envs/rapids-dask:
#
# Name                    Version                   Build  Channel
cudatoolkit               11.5.0               h36ae40a_9    nvidia
cupy-cuda115              9.6.0                    pypi_0    pypi
dask                      2021.11.2+18.g43247806           dev_0    <develop>
dask-cuda                 22.2.0a0+41.g2e9a429           dev_0    <develop>
dask-cudf                 22.2.0a0+156.g8c82d6acd1           dev_0    <develop>
dask-glm                  0.2.0                      py_1    conda-forge
dask-ml                   2021.11.30         pyhd8ed1ab_0    conda-forge
dask-sql                  2021.11.1.dev7+gebbfc2f           dev_0    <develop>
distributed               2021.11.2+19.gababd210           dev_0    <develop>
```
```
$ nvidia-smi
Tue Dec  7 10:42:23 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.00       Driver Version: 510.06       CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Quadro RTX 8000     On   | 00000000:15:00.0 Off |                  Off |
| 34%   32C    P8    17W / 260W |    444MiB / 49152MiB |     N/A      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  Quadro RTX 8000     On   | 00000000:2D:00.0  On |                  Off |
| 34%   60C    P0    70W / 260W |   1752MiB / 49152MiB |     N/A      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

</details>

I suspect that maybe this is just another issue with NVML in WSL2, similar to the one that came up in https://github.com/dask/distributed/issues/5342? Trying to run the bare pynvml code also fails:

```python
from pynvml import *

nvmlInit()

h = nvmlDeviceGetHandleByIndex(0)
nvmlDeviceGetComputeRunningProcesses_v2(h)
---------------------------------------------------------------------------
NVMLError_Unknown                         Traceback (most recent call last)
<ipython-input-8-e36e7bf91b7f> in <module>
----> 1 nvmlDeviceGetComputeRunningProcesses_v2(h)

~/miniconda3/envs/rapids-dask/lib/python3.8/site-packages/pynvml/nvml.py in nvmlDeviceGetComputeRunningProcesses_v2(handle)
   2139     else:
   2140         # error case
-> 2141         raise NVMLError(ret)
   2142 
   2143 def nvmlDeviceGetComputeRunningProcesses(handle):

NVMLError_Unknown: Unknown Error
```

**Comment by pentschev** [comment]
@charlesbluca do you know if there's a way to special-case WSL? We could then disable the check for existing CUDA contexts on WSL only to unblock this.

**Comment by charlesbluca** [comment]
Looks like `platform.uname` should be able to detect it? This is just from [a quick Google search](https://www.scivision.dev/python-detect-wsl/). there might be a better way to do this:

```python
from platform import uname

def in_wsl() -> bool:
    return 'microsoft-standard' in uname().release
```

I can play around with special casing `has_cuda_context` in Distributed, and also open a larger issue there since the root cause seems to lie there.

**Comment by charlesbluca** [comment]
Opened https://github.com/dask/distributed/pull/5568 to hopefully resolve this issue - @kuonumber, would you mind trying out this PR locally along with `DASK_DISTRIBUTED__DIAGNOSTICS__NVML=False` set to verify that this solves your problem?

**Comment by pentschev** [comment]
> Opened [dask/distributed#5568](https://github.com/dask/distributed/pull/5568) to hopefully resolve this issue - @kuonumber, would you mind trying out this PR locally along with `DASK_DISTRIBUTED__DIAGNOSTICS__NVML=False` set to verify that this solves your problem?

Please note that for that PR you'll need to be on RAPIDS 21.12, RAPIDS 21.10 will probably not work with that.

**Comment by kuonumber** [comment]
@pentschev Sad.. Option 1 and 2 can not work, or even I set DASK_DISTRIBUTED__DIAGNOSTICS__NVML=False.
One strange thing happened(Never get it while using older RAPIDS (like version21.06)...
![image](https://user-images.githubusercontent.com/16162246/145129348-2fee9ab5-6c49-4a7e-8938-ddb5d7425bfe.png)

**Comment by kuonumber** [comment]
@pentschev @charlesbluca 
I created three envs for testing, RAPIDS 21.12 with cudatoolkit 11.2, RAPIDS 21.10  with cudatoolkit 11.2, and RAPIDS 21.10  with cudatoolkit 11.0.
All work fine when I export DASK_DISTRIBUTED__DIAGNOSTICS__NVML=False.

![image](https://user-images.githubusercontent.com/16162246/145133181-632c9c35-264a-4ad0-be1d-9eb946c2f2d6.png)
![image](https://user-images.githubusercontent.com/16162246/145133278-a866ed1e-5bba-4724-a956-0f37095ce013.png)

**Comment by kuonumber** [comment]
Well, I think it can not work properly.

```
[dask_cudf.read_csv(f, chunksize="512 MiB") for f in filenames[-3:-1]]
# try to read two csv files and use larger chunk size.
```

No matter what chunksize I set, I stuck in MemoryError.
```
---------------------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
/tmp/ipykernel_20224/3369636072.py in <module>
----> 1 dask_cudf.concat([dask_cudf.read_csv(f, chunksize="512 MiB") for f in filenames[-3:-1]])

/tmp/ipykernel_20224/3369636072.py in <listcomp>(.0)
----> 1 dask_cudf.concat([dask_cudf.read_csv(f, chunksize="512 MiB") for f in filenames[-3:-1]])

~/miniconda3/envs/rapids21.10_cuda10/lib/python3.7/site-packages/dask_cudf/io/csv.py in read_csv(path, chunksize, **kwargs)
     67         return func(path, blocksize=chunksize, **kwargs)
     68     else:
---> 69         return _internal_read_csv(path=path, chunksize=chunksize, **kwargs)
     70 
     71 

~/miniconda3/envs/rapids21.10_cuda10/lib/python3.7/site-packages/dask_cudf/io/csv.py in _internal_read_csv(path, chunksize, **kwargs)
    109 
    110     if chunksize is None:
--> 111         return read_csv_without_chunksize(path, **kwargs)
    112 
    113     dask_reader = make_reader(cudf.read_csv, "read_csv", "CSV")

~/miniconda3/envs/rapids21.10_cuda10/lib/python3.7/site-packages/dask_cudf/io/csv.py in read_csv_without_chunksize(path, **kwargs)
    164     # Read "head" of first file (first 5 rows).
    165     # Convert to empty df for metadata.
--> 166     meta = cudf.read_csv(filenames[0], nrows=5, **kwargs).iloc[:0]
    167 
    168     graph = {

~/miniconda3/envs/rapids21.10_cuda10/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

~/miniconda3/envs/rapids21.10_cuda10/lib/python3.7/site-packages/cudf/io/csv.py in read_csv(filepath_or_buffer, lineterminator, quotechar, quoting, doublequote, header, mangle_dupe_cols, usecols, sep, delimiter, delim_whitespace, skipinitialspace, names, dtype, skipfooter, skiprows, dayfirst, compression, thousands, decimal, true_values, false_values, nrows, byte_range, skip_blank_lines, parse_dates, comment, na_values, keep_default_na, na_filter, prefix, index_col, **kwargs)
    108         na_filter=na_filter,
    109         prefix=prefix,
--> 110         index_col=index_col,
    111     )
    112 

cudf/_lib/csv.pyx in cudf._lib.csv.read_csv()

MemoryError: std::bad_alloc: CUDA error at: /home/kuo/miniconda3/envs/rapids21.10_cuda10/include/rmm/mr/device/cuda_memory_resource.hpp:70: cudaErrorNotSupported operation not supported
```

**Comment by kuonumber** [comment]
By the way, after upgrading  RAPIDS from 21.06 to 21.10 or 21.12, 
An error occurred. 
![image](https://user-images.githubusercontent.com/16162246/145137743-7127e5b9-cf2e-4ae7-9810-da9e039f604c.png)

**Comment by pentschev** [comment]
The `MemoryError` reported in https://github.com/rapidsai/dask-cuda/issues/816#issuecomment-988435990 is very likely getting due to the GPU being out of memory and not related to the original issue of this thread. I would suggest trying a very small dataset first, perhaps a 2GB dataset first. As I mentioned previously Dask-CUDA is not optimized for supporting GPUs with different memory sizes, so you'll likely hit issues when the smallest GPU runs out of memory even if the largest GPU would be able to fit the data.

The error with `mem_info` seems the same limitation with WSL for which we have to disable NVML in Distributed. Is that error coming for Dask or is it your own script @kuonumber ? I suspect it's the latter because I don't think Dask uses CuPy at all to query GPU memory info, so for Dask even if that isn't supported by CUDA in WSL it should be ok.

**Comment by charlesbluca** [comment]
@kuonumber I am able to run your example CuPy code on WSL using CUDA toolkit 11.0 and 11.2 - it's possible that your NVIDIA driver installation may be either outdated or incorrect - for a sanity check, the latest drivers for WSL can be found [here](https://developer.nvidia.com/cuda/wsl/download).

I would recommend reopening this particular issue in [CuPy](https://github.com/cupy/cupy/issues), as there are likely more folk there who have the knowledge to help you with this problem.

**Comment by kuonumber** [comment]
@charlesbluca But this code can run on RAPIDS 21.06  with CUDA toolkit 11.0 or 11.2.  And I installed driver from the same source. 
Ok. I will send it to Cupy repo. Thank you.

**Comment by kuonumber** [comment]
@pentschev 
Think you .
You are right, **mem_info** is not from dask. 

For the **MemoryError  part,**
I tried two ways. 
experiment  1 is to use small enough csv.
experiment  2 is to use single gpu 1080ti on cluster mode.
Unfortunately, both return same error again.

**Comment by pentschev** [comment]
> For the MemoryError part,
I tried two ways.
experiment 1 is to use small enough csv.
experiment 2 is to use single gpu 1080ti on cluster mode.
Unfortunately, both return same error again.

That's really strange, but I now noticed the `MemoryError` is not OOM, but rather `cudaErrorNotSupported operation not supported`. So it seems that something is still not working right. What RAPIDS version were you using when that error was shown? It seems to me that this was only with RAPIDS 21.10, so it would be useful to know whether the same error happens with RAPIDS 21.12.

@charlesbluca do you think you could try to reproduce that by trying some `dask_cudf.read_csv` calls?

**Comment by charlesbluca** [comment]
Sure, I can try to make a small reproducer on 21.10 / 21.12 - for reference, I noticed earlier that cuDF's reader / writer functions were segfaulting for WSL in the latest source and opened issues for that in https://github.com/rapidsai/cudf/issues/9859 https://github.com/rapidsai/cudf/issues/9861

**Comment by charlesbluca** [comment]
@kuonumber, for your `read_csv` operations, are you attempting to read compressed CSV files? I notice from your stacktrace that your `chunksize` is ignored, which seems to occur when compressed CSVs are used:

https://github.com/rapidsai/cudf/blob/e05bd4bf3cf410058417268e4353de46d387ac96/python/dask_cudf/dask_cudf/io/csv.py#L99-L108

**Comment by charlesbluca** [comment]
So far I haven't been able to reproduce any of the CUDA level memory issues 😕 my best guess here is that these issues are coming up as a result of your eGPU, which to my knowledge doesn't have any documentation as to which queries are and aren't supported.

I notice in https://github.com/rapidsai/dask-cuda/issues/816#issuecomment-988441768 that you are attempting to query GPU 1, which I assume is the eGPU? As a sanity check, could you try the same query on your internal GPU?

```python
import cupy

cupy.cuda.Device(0).mem_info
```

That should help narrow down if this issue is specific to one of your GPUs.

**Comment by kuonumber** [comment]
@pentschev 
I tried both (RAPIDS 21.10 and RAPIDS 21.12), and got same error.

**Comment by kuonumber** [comment]
@charlesbluca 
I stopped my egpu, and used the internal one.
Either cudf.read_csv() or dask_cudf.read_csv() can not read **not compressed** csv.

![image](https://user-images.githubusercontent.com/16162246/145743993-541c576d-fadb-47ab-8c2e-7c1f544a7e8b.png)

**Comment by charlesbluca** [comment]
Really sorry for the delayed response here @kuonumber - the fact that `cupy.cuda.Device(0).mem_info` is failing here seems to be an indication that there might be something wrong with your driver / toolkit installation.

I would recommend upgrading to your most recent [NVIDIA drivers](https://www.nvidia.com/Download/index.aspx?lang=en-us) (production branch), installing a fresh WSL2 distro (`wsl --install -d <Distro>`), installing Conda + CuPy in that distro and checking to see if `mem_info` works there - if it does, then we can isolate the issues here to something in your current WSL2 distro. If not, then hopefully we can parse out if the issues here have something to do with your specific GPUs.

**Comment by pentschev** [comment]
I believe the issues here are likely fixed in latest Dask/Distributed/NVIDIA driver. I'm tentatively closing this since it has gone stale, please feel free to reopen if those are still an issue after upgrading.
